service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/GcsBucketCloudSyncStep.java

```diff
@@ -45,32 +48,41 @@ public StepResult doStep(FlightContext flightContext)
+      GcpPolicyBuilder updatedPolicyBuilder =
+          new GcpPolicyBuilder(resource, projectId, currentPolicy);
 
-    // Read Sam groups for each workspace role.
-    Map<WsmIamRole, String> workspaceRoleGroupsMap =
-        workingMap.get(WorkspaceFlightMapKeys.IAM_GROUP_EMAIL_MAP, new TypeReference<>() {});
-    for (Map.Entry<WsmIamRole, String> entry : workspaceRoleGroupsMap.entrySet()) {
-      updatedPolicyBuilder.addWorkspaceBinding(entry.getKey(), entry.getValue());
-    }
+      // Read Sam groups for each workspace role.
+      Map<WsmIamRole, String> workspaceRoleGroupsMap =
+          workingMap.get(WorkspaceFlightMapKeys.IAM_GROUP_EMAIL_MAP, new TypeReference<>() {});
+      for (Map.Entry<WsmIamRole, String> entry : workspaceRoleGroupsMap.entrySet()) {
+        updatedPolicyBuilder.addWorkspaceBinding(entry.getKey(), entry.getValue());
+      }
 
-    // Resources with permissions given to individual users (private or application managed) use
-    // the resource's Sam policies to manage those individuals, so they must be synced here.
-    // This section should also run for application managed resources, once those are supported.
-    if (resource.getAccessScope() == AccessScopeType.ACCESS_SCOPE_PRIVATE) {
-      Map<ControlledResourceIamRole, String> resourceRoleGroupsMap =
-          workingMap.get(
-              ControlledResourceKeys.IAM_RESOURCE_GROUP_EMAIL_MAP, new TypeReference<>() {});
-      for (Map.Entry<ControlledResourceIamRole, String> entry : resourceRoleGroupsMap.entrySet()) {
-        updatedPolicyBuilder.addResourceBinding(entry.getKey(), entry.getValue());
+      // Resources with permissions given to individual users (private or application managed) use
+      // the resource's Sam policies to manage those individuals, so they must be synced here.
+      // This section should also run for application managed resources, once those are supported.
+      if (resource.getAccessScope() == AccessScopeType.ACCESS_SCOPE_PRIVATE) {
+        Map<ControlledResourceIamRole, String> resourceRoleGroupsMap =
+            workingMap.get(
+                ControlledResourceKeys.IAM_RESOURCE_GROUP_EMAIL_MAP, new TypeReference<>() {});
+        for (Map.Entry<ControlledResourceIamRole, String> entry :
+            resourceRoleGroupsMap.entrySet()) {
+          updatedPolicyBuilder.addResourceBinding(entry.getKey(), entry.getValue());
+        }
       }
-    }
 
-    logger.info(
-        "Syncing workspace roles to GCP permissions on bucket {}", resource.getBucketName());
-    wsmSaStorageCow.setIamPolicy(resource.getBucketName(), updatedPolicyBuilder.build());
+      logger.info(
+          "Syncing workspace roles to GCP permissions on bucket {}", resource.getBucketName());
+
+      wsmSaStorageCow.setIamPolicy(resource.getBucketName(), updatedPolicyBuilder.build());
+    } catch (StorageException e) {
+      if (e.getCode() == HttpStatus.BAD_REQUEST.value()) {
```

Comment:
Could you also add a comment explaining why we retry 400s here for the future? It's a little unexpected at first glance